src/embed.js

```diff
@@ -68,11 +76,25 @@ function embed(el, opt, callback) {
       source = JSON.stringify(spec, null, 2);
       opt = {spec: spec, actions: false};
     }
-    mode = MODES[opt.mode] || MODES.vega;
+
+    // Decide mode
+    if (opt.mode) {
+      mode = MODES[opt.mode]
+    } else if(spec.$schema) {
+      const parsed = ural_parser(spec.$schema);
+      mode = MODES[parsed.library];
+      if (parsed.version !== VERSION[mode]){
```

Comment:
This seems to not work correctly yet. 

<img width="698" alt="screen shot 2017-02-04 at 13 31 21" src="https://cloud.githubusercontent.com/assets/589034/22621680/497f2006-eade-11e6-871f-219c0ce2071e.png">

Maybe the easiest way is to simply import https://github.com/npm/node-semver